### DISCUSSION TITLE
V1.4.0: calibration not possible

### DISCUSSION THREAD

**Discussion by maibaam** [mp-se/gravitymon#177] [Q&A]
Hello, 

I have problems to get the gravitymon working.

But first: thank you for your amazing work on developping this project and sharing it on github! 

I have an iSpindel-device, that primarily had installed the original iSpindel-firmware (universam) and was calibrated. It worked and I was able to get all data the device is made for. 
Because I find the gravitymon-project more useful for my needs and - anyway - had to perform a calibration, I decided to switch to gravitymon-firmware and flashed the device with v1.4.0. 

First issue I had to observe:
The data already stored in the former firmware (calibration, polynominal, WiFi-settings, ...) was not migrated, though I used the GUI of the former firmware to initiate the firmware transition.
As I had read before in the instructions on the gravitymon-website, this way should have kept and migrated the former data, that was already gathered.
As instructed, I did not do a full erase before flashing. 
 
Thus, I was forced to newly set up all data from scratch or respectively filled in some fields (e.g. for testing: former polynominal) with data I had written down before flashing with gravitymon-firmware.

Next issues I had to observe:
It's not possible to calibrate the device (aligned horizontally). 
When clicking on the button "Calibrate device", nothing happens. 

The buttons "Gather support Information" and "submit an issue on github" also do not work (that's why I cannot provide them).

Does anyone have a hint, what I can do to come along with calibration?

Thanks, Peter.

**Comment by mp-se** [comment]
You are correct that the function is actually broken in the 1.4 version.  It looks like it broke when I moved to async webserver so if you use beta 1 to do the migration and then upgrade to the release that should work for now. I will fix the bug and release an update.

**Comment by mp-se** [comment]
You can try this build and see if that fixes the problem: https://github.com/mp-se/gravitymon/blob/dev/bin/firmware.bin

**Comment by gabeweisz-oldacct** [comment]
Both the 1.4.0 release and this build are very broken for me - the angle is totally off on ESP8266.

**Reply by mp-se** [reply]
Then its probably some of the gyro values that have been mixed up. Can you record the values after the ispindle migration and then do a 90 degree calibration and record the new values? 

The values are displayed alongside the calibration button.

**Reply by maibaam** [reply]
I assume, gabeweisz encountered the same behaviour as I did (and as the title says) - pressing the calibration button doesn't result in anything. The "new" values all remain "0".

**Reply by mp-se** [reply]
> I assume, gabeweisz encountered the same behaviour as I did (and as the title says) - pressing the calibration button doesn't result in anything. The "new" values all remain "0".

Yes the feature is broken, i never delete the ispindel config so the data is not lost. I will do some more tests tonight to see if i migrate the values correcly or not.

**Reply by mp-se** [reply]
Did you restart the device after migration ? this is needed for the gyro calibration to be applied. This was not really clear in the docs/information.

**Reply by maibaam** [reply]
Me (at least and of course) rebooted the device once I had etablishes major changes. But please also view my procedure under the next comment...

**Comment by maibaam** [comment]
> You can try this build and see if that fixes the problem: https://github.com/mp-se/gravitymon/blob/dev/bin/firmware.bin

Hello Magnus,

thank you for providing firmware v1.4.1 under https://github.com/mp-se/gravitymon/blob/dev/bin/firmware.bin

For me, it worked now as follows:

0. coming from the not working calibration tool with gravitymon v1.4.0 beeing installed:
1. via the gravitymon-menu "_configuration > upload firmware_": first reinstall original iSpindel-firmware (from github, universam, release 7.3.3)
2. level the device horizontally (by using a calibrated tablet as a 2-dimensional waterlevel)
3. perform a calibration (under iSpindel firmware)
4. via iSpindel-menu "_maintenance > update_": install gravitymon v1.4.1
5. Reboot the device.
6. When starting up the gravitymon webserver, it (now) indicated that it found a preset of values from the former (abovementioned) iSpindel (that did not work in v1.4.0).
(I did not know that you keep the iSpindel data/values on the flash while installing gravitymon (though we did not indicate to not do a ful erase) - so I first worried about how to avoid erasing the flash that - as you mentioned - keeps the former values. But: does this not restrict the available memory for other operations, what is crucial with an ESP8266 ?)
7. Tried a new calibration via gravitymon-menu "_configuration > (button) Calibrate device_" --> this time, even this worked!
As a hint: in my case, the following values (what you asked for) are reported: "_-574,-36,1012,707,-49,-20 88.664_"
8. Thereafter, I directly tested a value (tilt) to be found in distilled water; that resulted in SG 1,0025 (~0,64 °P). As a start, that should be sufficient to me; further formula calculation can be done later on (besides: it is very useful, that you managed to store once entered values in the calculation table; so one does not have do write them down separately over time when finding out further benchmarks).

Some infos that might be useful for your further development:
I tested the functionality and delivered values with MQTT and nodered (besides: the format editor is quite helpful):
- when using the DS18B20, the temperature first was reported with no position after decimal point, then 1 position;
- when setting the device to use the gyro-temperature-sensor, it is reported with 2 positions after decimal point;
- voltage is reported as V --> I will use a math node in nodered to convert it to percent according to the specific battery;
- Until now, I found an _average runtime_ of 6 seconds; though RSSI at -55... (you wrote about an optimum at < 1 s)? Unloads the battery?
- In gravitymon-menu "_configuration > device_", the buttons "_Submit an issue on github_" and "_Gather support information_" still do not work.

Thank you so long!
Cheers, Peter.

**Reply by mp-se** [reply]
The feature works in this way.

When you install gravmon via the ispindel upload feature the file system is kept intact (a full erase does a filesystem format as well). Gravmon checks if there is an ispindel configuration file and if so extracts the values from that and then saves those in its own configuration file. If that is successful then the ispindel file is renamed from config.json to ispindel.json. There is plenty of space in the file system so this is not a big deal. You can always export the configuration and do a flash with erase to clear the whole flash memory + restore of configuration. This has no impact on the RAM memory. 

>when using the DS18B20, the temperature first was reported with no position after decimal point, then 1 position;

The number of decimals for the temperature should be consistent, was this on the index page in the web UI? If the value was just zeros after the decimal point it might skip the decimals... but that can be fixed. 

>voltage is reported as V --> I will use a math node in nodered to convert it to percent according to the specific battery;

There is an option to report battery as percentage in the format templates.

>Until now, I found an average runtime of 6 seconds; though RSSI at -55... (you wrote about an optimum at < 1 s)? Unloads the >battery?

1 second is quite optimistic, should be less than 10 seconds...  I will check the docs

>In gravitymon-menu "configuration > device", the buttons "Submit an issue on github" and "Gather support information" still do not work.

I will have a look at that part as well before releasing the new version.
